### ISSUE TITLE
Implement CSS cascading

### ISSUE THREAD

**Issue by lehni** [jsdom/jsdom#1696]
Am I doing something wrong, or is JSDOM completely lacking support for cascading inline styles?

A simple example:

```js
var doc = jsdom.jsdom('<html><body><div style="color: red;"><div id="nested"></div></div></body></html>');
var win = doc.defaultView;
var el = doc.getElementById('nested');
console.log(win.getComputedStyle(el, ''));
```

Output:
```
CSSStyleDeclaration {
  '0': 'display',
  _values: { display: 'block' },
  _importants: { display: '' },
  _length: 1,
  _onChange: [Function] }
```

As reference, here the browser's behavior for the same example:

https://jsfiddle.net/qxu1v1zc/

**Comment by Sebmaster** [comment]
> is JSDOM completely lacking support for cascading inline styles?

That is true afaik. No one has written anything to properly deal with cascading yet.

**Comment by lehni** [comment]
Ouch... That sounds like a massive task? As far as style handling goes, it alos looks like there isn't any kind of style parsing in place in `getComputedStyle()`? For example, the browser returns `rgb(255, 0, 0)` for the color in the above example, while jsdom returns `red`.

**Comment by TidyIQ** [comment]
Any idea when/if this will be fixed? My tests are failing due to getComputedStyle failing due to lack of cascading in JSDOM.

**Comment by domenic** [comment]
https://mobile.twitter.com/slicknet/status/782274190451671040

**Comment by domenic** [comment]
**Posted by GPT-6 Astra Extra High**

The inline color-inheritance case in this report is fixed. The enabled [CSS Color inheritance WPT](https://github.com/web-platform-tests/wpt/blob/e94af787e39d4161f01f0ef78d933f1103ee32b4/css/css-color/inheritance.html) uses [the inheritance helper](https://github.com/web-platform-tests/wpt/blob/e94af787e39d4161f01f0ef78d933f1103ee32b4/css/support/inheritance-testcommon.js) to set a parent’s inline color and verify the child’s computed color follows it. It is [enabled in jsdom’s WPT suite](https://github.com/jsdom/jsdom/blob/7a3ea0bfff1d1bc61c8f112304914eb20a4de78c/test/web-platform-tests/to-run.yaml#L285) and passes on current main with no skip or failing expectation. Closing this reported inheritance bug; this does not imply that every CSS cascade feature is implemented.
